pandas/core/generic.py

```diff
@@ -6182,6 +6182,28 @@ def __finalize__(self, other, method: str | None = None, **kwargs) -> Self:
             )
             self.flags.allows_duplicate_labels = allows_duplicate_labels
 
+        if method == "groupby" and len(self) > 0:
+            for i, col in enumerate(self.columns):
+                master_scalar = next(
+                    value for value in self.iloc[:, i].values if not isna(value)
+                )
+                if hasattr(master_scalar, "from_sequence"):
+                    try:
+                        from pint_pandas import PintType
```

Comment:
oh goodness, no.  ill have to look at the original issue(s), but special-casing 3rd party EAs is definitely not a pattern we want (im actively trying to get rid of places where we special case our own EAs!)

Reply:
I totally get it (and the reason I mentioned it as a problem in the commit log).  I initially tried to write this up as an Issue for discussion, but the templates offered were all so normative in unhelpful ways that I thought it would be better to start the discussion as a PR.  That may have been a bit rash...

Reply:
> but the templates offered were all so normative in unhelpful ways that I thought it would be better to start the discussion as a PR

Totally reasonable.  The templates are mostly to guide first-time contributors, so at this point no one will complain if you ignore the template at your own discretion.